### PR TITLE
Adds basic enemy AI and begins combat

### PR DESCRIPTION
**Pull request by ZayahPapaya**

### CODE REVIEW

File: src/components/GameBoard/GameBoard.tsx [hunk 1]
```diff
@@ -80,6 +70,18 @@ const Board = (props: BoardProps) => {
...
   }
 
+  const doCombat = (e: any, entity: Entity) => {
+    const zombie = new Enemy('Zombie', racc, 10, 5)
+    moveEntity(zombie, [entity.position![0] + 1, entity.position![1]])
+    if(entity instanceof Raccoon) {
+    const raccoon = entity as Raccoon;
+    raccoon.changeWeapon(new Axe());
+    raccoon.useWeapon();
+    };
```
Comment (laptopadventure) [inline src/components/GameBoard/GameBoard.tsx:80]: indent

File: src/components/GameBoard/GameBoard.tsx [hunk 2]
```diff
@@ -80,6 +70,18 @@ const Board = (props: BoardProps) => {
     setCurrentEntities([...currentEntities]);
   }
 
+  const doCombat = (e: any, entity: Entity) => {
```
Comment (laptopadventure) [inline src/components/GameBoard/GameBoard.tsx:73]: use the event or remove the arg but make sure to type it

File: src/components/GameBoard/GameBoard.tsx [hunk 3]
```diff
@@ -71,7 +61,7 @@ const Board = (props: BoardProps) => {
   function debugPosition(event: React.MouseEvent<HTMLImageElement>, entity: Entity) {
     // console.log(entity.position)
     const { marginLeft, marginTop } = event.currentTarget.style
-    console.log(`${tilePx}:`, `${marginLeft}, ${marginTop}`)
+    //console.log(`${tilePx}:`, `${marginLeft}, ${marginTop}`)
```
Comment (laptopadventure) [inline src/components/GameBoard/GameBoard.tsx:64]: remove commented code

File: src/classes/entity.ts [hunk 4]
```diff
@@ -197,10 +192,60 @@ export class Raccoon extends Mob {
...
+      console.log(`${this.name} is attacking ${solid.name}`);
+      this.attack(solid);
+      return;
+    }// ^ attacks the solid if one was found
+
+    return this.moveToPosition(targetTile);
+    // ^ move to tile if unoccupied
+  };
+}
+//return this.moveToPosition(targetTile); // if tile is empty, move to tile
```
Comment (laptopadventure) [inline src/classes/entity.ts:248]: remove commented code

**Review by laptopadventure — CHANGES_REQUESTED** [review]

**Review by JimD325 — APPROVED** [review]

**Review by laptopadventure — APPROVED** [review]

**Pull request merged**